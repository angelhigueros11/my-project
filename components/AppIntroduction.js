import React from 'react';
import { StyleSheet, Text, View,SafeAreaView,Image,TextInput,TouchableOpacity,Pressable,Alert,ActivityIndicator} from 'react-native';
import LottieView from 'lottie-react-native';

export default function AppIntroduction({navigation}){
    return (
        <SafeAreaView style={styles.container}>
           <View style={styles.logo}>
                <Text style={styles.logoText1}>Health</Text>
                <Text style={styles.logoText2}>Forms</Text>
            </View>
            <LottieView
                source={require('../assets/Lottie/clouds.json')}
                autoPlay={true}
                loop={true}
                speed={0.5}
                style={styles.nubes}
            
            />
            <LottieView
                    source={require('../assets/Lottie/doctor.json')}
                    autoPlay={true}
                    loop={true}
                    style={styles.animation}
                />
            <View style={styles.textContainer}>

                <View style={styles.mainContainer}>
                    
                    <View style={styles.proyectDescriptionsContainer}>
                        <Text style={styles.proyectDescription}>
                        Esta aplicaci칩n busca impactar a la poblaci칩n guatemalteca en cuanto la agilizaci칩n
                        de tr치mites de la salud (llenado de formularios)
                        </Text>

                    </View>

                    <TouchableOpacity style={styles.logInButton} activeOpacity={0.8} onPress={() => navigation.navigate("UserLogin")}>
                            <Text style={{color:'#29abe2'}}>Iniciar</Text>
                    </TouchableOpacity>

                </View>
               
            </View>
            
        </SafeAreaView>
    )
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'flex-start',
        alignItems: 'center',
        backgroundColor:'#29abe2',
        width: '100%',
    },
    logo:{
        flex: 0.2,
        flexDirection: 'row',
        alignItems:'flex-end',
        justifyContent: 'center',
        marginTop:0
      },
      logoText1:{
        fontSize:40,
        color: '#fff',
        fontWeight: 'bold',
      },
      logoText2:{
        fontSize:40,
        fontWeight: 'bold',
        color: '#fff',


      },
      nubes:{
          marginTop:50,
      },
      animation:{
          position: 'absolute',
          marginTop:0
          
          
      },textContainer:{
          flex: 1,
        //   backgroundColor: '#fff',
          justifyContent: 'flex-end',
          width: '90%',

      },mainContainer:{
          flex: 0.43,
          backgroundColor: '#29abe2',
          width: "100%",
          borderTopRightRadius:30,
          borderTopLeftRadius:30,
          justifyContent: 'flex-start',
          alignItems: 'center',
          
      },proyectDescriptionsContainer:{
          height:'auto',
          width: "70%",
        //   backgroundColor: 'red',
          marginTop: 50,
          marginLeft:0, 
      },
      proyectDescription:{
        fontSize:15,
        // fontWeight: 'bold',
        textAlign: 'center',
        color: '#fff',
      },
      logInButton:{

        width:"80%",
        height:40,
        borderRadius:15,
        backgroundColor: '#fff',
        justifyContent: 'center',
        alignItems: 'center',
        position:'relative',
        top:20,
        color: '#29abe2',
    }

})