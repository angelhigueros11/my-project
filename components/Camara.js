import React, { useState, useEffect } from 'react';
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native';
import { Camera } from 'expo-camera';
import { Ionicons } from '@expo/vector-icons'; 
import { BarCodeScanner } from 'expo-barcode-scanner';

export default function Camara({navigation}) {
  const [hasPermission, setHasPermission] = useState(null);
  const [scanned, setScanned] = useState(false);
  const [text, setText] = useState("Vacio");
  // const [type, setType] = useState(Camera.Constants.Type.back);

  const askForCamera = () => {
    (async () => {
      const {status} = await BarCodeScanner.requestPermissionsAsync();
      setHasPermission(status=="granted");
    })()
  }

  useEffect(() => {
    askForCamera();
  }, []);


  const handleBarcode = ({type, data}) => {
    var alphabet = "abcdefghijklmnopqrstuvwxyz0123456789[]{}";
    var decifrado = "";
    for (var i = 0; i < data.length; i++) {
      if(alphabet.includes(data[i].toLocaleLowerCase())){
        let pos = alphabet.indexOf(data[i].toLocaleLowerCase());
        decifrado += alphabet[(((pos-5)%alphabet.length) + alphabet.length) % alphabet.length]
      }else {decifrado += data[i].toLocaleLowerCase();}
    }
    // console.log(decifrado)
    setScanned(true);
    const parsedData = JSON.parse(decifrado);
    const name = parsedData.name;
    const lastName = parsedData.lastname;
    const age = parsedData.age;
    const document = parsedData.document;
    const birth = parsedData.birth;
    const smoker = parsedData.smoker;
    const med = parsedData.med;
    const pregnant = parsedData.pregnant;
    const titulo = parsedData.titulo;
    const fechaCreacion = parsedData.fechacreacion;
    navigation.navigate("UserInfoView", {name,lastName,age,document,birth,smoker,med,pregnant,titulo,fechaCreacion})
  }


  if (hasPermission === null) {
    return (
      <View style={styles.containerNull}>
         <Text style={{color: 'white', fontSize:20}}>Solicitando permiso de camara</Text>
      </View>
    );
  }
  if (hasPermission === false) {
    return (
      <View style={styles.containerNull}>
         <Text style={{color: 'white', fontSize:20}}>Sin acceso a la camara</Text>
         <TouchableOpacity style={styles.scanButton} onPress={()=>{askForCamera()}}>
          <Text style={styles.scanButtonText}>Dar acceso a la camara</Text>
        </TouchableOpacity>
      </View>
    );
  }
  return (
    <View style={styles.containerNull}>
      <View style={styles.BarCodeScanner}>
        <BarCodeScanner
        onBarCodeScanned={scanned ? undefined : handleBarcode}
        style={{height:900,width:200}}/>
      </View> 
      {/* <Text style={styles.pTexto}>{text}</Text> */}

      {/* {scanned && 
      <TouchableOpacity style={styles.scanButton} onPress={()=>{setScanned(false)}}>
          <Text style={styles.scanButtonText}>Escanear otra vez</Text>
        </TouchableOpacity>} */}
      
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#143590',
    
  },
  BarCodeScanner: {
    alignItems:"center",
    justifyContent: "center",
    height:200,
    width:200,
    borderRadius:30,
    overflow:"hidden",
    backgroundColor: 'red'
  },pTexto:{
    fontSize: 18,
    color: 'white'
  },
  containerNull:{
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#143590',
    alignItems: 'center',
  },
  camera: {
    flex: 1,
  },
  buttonContainer: {
    flex: 1,
    backgroundColor: 'transparent',
    flexDirection: 'row',
    margin: 20,
    alignItems: 'flex-end',
    justifyContent: 'space-around',
  },icon: {
      marginBottom:15,
      alignSelf:"center",
  },
  shareButton:{
    marginTop:20,
    backgroundColor:"#143590",
    justifyContent:"center",
    alignItems: "center",
    width:120,
    height:50,
    alignSelf: "center",
    borderRadius:15,
    alignSelf:"flex-end"

},
shareButtonText:{
    color:"white",
},
  button: {
    flex: 0.1,
    flexDirection: 'row',
    alignSelf: 'flex-end',
    alignItems: 'center',
    width:"80%",

   
  },
  text: {
    fontSize: 18,
    color: 'white',
  },scanButton:{
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#0657a2',
    borderRadius:8,
    height: 50,
    marginTop:25,
    bottom:10,
    width:'60%',
    alignSelf: 'center',
    
  },scanButtonText:{
    color: 'white',
    fontSize:15,
    marginLeft:10
  }
});