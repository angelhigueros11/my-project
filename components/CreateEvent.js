import React, { useState } from "react";
import {
  Alert,
  StyleSheet,
  Text,
  TextInput,
  View,
  TouchableOpacity,
  Platform,
  ScrollView,
} from "react-native";
import firebase from "./firebase2.js";

function saveEvent(navigation, email, title, date, descripcion) {
  if (
    title !== "" &&
    email !== "" &&
    descripcion !== "" &&
    Date.parse(date) !== NaN
  ) {
    try {
      firebase.db.collection(email + "historialMedico").add({
        titulo: title,
        descripcion: descripcion,
        fecha: date,
        creationDate: new Date().toLocaleDateString(),
      });

      Alert.alert("Formulario", "Se creo el formulario");
      navigation.goBack();
    } catch (err) {
      Alert.alert("Error", "Algo salio mal, intentalo m√°s tarde");
      navigation.goBack();
    }
  } else {
    Alert.alert(
      "Datos invalidos",
      "Asegurese que todos los campos estan llenos en el formato correcto"
    );
  }
}

export default function CreateEvent({ navigation, route }) {
  const [titulo, setTitulo] = useState("");
  const [fecha, setFecha] = useState("");
  const [descripcion, setDescripcion] = useState("");
  const email = route.params.email;

  const [buttonEnabled, setButtonEnabled] = useState(true);
  return (
    <View style={styles.container}>
      <View
        behavior={Platform.OS === "ios" ? "padding" : "padding"}
        style={styles.container}
      >
        <View style={styles.container}>
          <View style={styles.infoContainer}>
            <ScrollView style={styles.infoScroll}>
              <View style={styles.infoContainer2}>
                <View style={styles.row}>
                  <Text>Titulo: </Text>
                  <TextInput
                    style={[
                      styles.input,
                      { color: buttonEnabled ? "black" : "#c9c9c9" },
                    ]}
                    placeholder="Titulo"
                    value={titulo}
                    onChangeText={(text) => setTitulo(text)}
                    autoCapitalize="none"
                    editable={buttonEnabled}
                  />
                </View>

                <View style={styles.row}>
                  <Text>Fecha: </Text>
                  <TextInput
                    style={[
                      styles.input,
                      { color: buttonEnabled ? "black" : "#c9c9c9" },
                    ]}
                    placeholder="Fecha evento"
                    value={fecha}
                    onChangeText={(text) => setFecha(text)}
                    autoCapitalize="none"
                    editable={buttonEnabled}
                  />
                </View>
                <View
                  style={[
                    styles.row,
                    {
                      justifyContent: "flex-start",
                      marginLeft: 32,
                      marginTop: 35,
                    },
                  ]}
                >
                  <Text>Descripcion: </Text>
                </View>
                <View
                  style={[
                    styles.row,
                    { justifyContent: "flex-start", marginLeft: 32 },
                  ]}
                >
                  <TextInput
                    style={[
                      styles.input,
                      {
                        color: buttonEnabled ? "black" : "#c9c9c9",
                        borderBottomWidth: 0,
                        height: 100,
                        width: "80%",
                      },
                    ]}
                    placeholder="Descripcion"
                    value={descripcion}
                    onChangeText={(text) => setDescripcion(text)}
                    autoCapitalize="none"
                    editable={buttonEnabled}
                    multiline
                  />
                </View>
              </View>
            </ScrollView>
          </View>
        </View>

        <View style={styles.buttonArea}>
          <TouchableOpacity
            style={styles.shareButton}
            onPress={() => {
              saveEvent(navigation, email, titulo, fecha, descripcion);
            }}
          >
            <Text style={styles.shareButtonText}>Crear</Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#f5f5f5",
    alignContent: "center",
    justifyContent: "flex-start",
    margin: 10,
  },
  shareButton: {
    marginTop: 20,
    backgroundColor: "#143590",
    justifyContent: "center",
    alignItems: "center",
    width: "30%",
    height: 50,
    alignSelf: "center",
    borderRadius: 15,
    margin: 20,
  },
  shareButtonText: {
    color: "white",
  },
  header: {
    flex: 0.1,
    backgroundColor: "#29abe2",
    width: "100%",
    flexDirection: "row",
    justifyContent: "flex-end",
    alignItems: "center",
  },
  content: {
    flex: 1,
    backgroundColor: "lightblue",
  },
  footer: {
    flex: 0.04,
    backgroundColor: "lightgrey",
  },
  formatexto: {
    flex: 1,
    fontSize: 25,
    fontStyle: "normal",
    fontWeight: "bold",
    color: "#FFFF",
    textAlign: "center",
  },
  formatexto2: {
    flex: 0.1,
    fontSize: 12,
    color: "black",
    textAlign: "left",
    margin: 15,
  },
  formatexto3: {
    flex: 1,
    fontSize: 10,
    fontWeight: "bold",
    color: "grey",
    textAlign: "center",
    margin: 3,
  },
  infoContainer: {
    flex: 1,
    width: "100%",
  },
  infoContainer2: {
    flex: 1,
    justifyContent: "center",
    // backgroundColor:'red',
    width: "100%",
  },
  infoScroll: {
    flex: 1,
    backgroundColor: "#fefefe",
  },
  input: {
    backgroundColor: "#fff",
    marginTop: 0,
    height: 40,
    borderBottomColor: "#143590",
    borderBottomWidth: 1,
    width: "50%",
    marginLeft: 5,
    marginTop: 10,
  },
  buttonArea: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  row: {
    flexDirection: "row",
    alignItems: "baseline",
    justifyContent: "space-evenly",
    marginTop: 12,
    marginBottom: 12,
    paddingHorizontal: 10,
  },
});
