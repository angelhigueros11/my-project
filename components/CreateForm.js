import { StatusBar } from 'expo-status-bar';
import React,{useState} from 'react';
import { KeyboardAvoidingView,Alert, StyleSheet, Text,TextInput, View,SafeAreaView,Image,TouchableOpacity,Platform,ScrollView } from 'react-native';
import { getDatabase, ref, set } from "firebase/database";
import firebase from './firebase.js';
// import firebase from './firebase2.js';

function saveFormInfo(navigation,email,title,name,lastName,age,document,birth,smoker,medicamento,pregnant,f){
    const TT = title;

    if(title!=='' && name!=='' && lastName!=='' && parseInt(age)!==NaN && smoker!=='' && document!=='' && Date.parse(birth)!==NaN){
        try{
            if(f){
                firebase.db.collection(email).add(
                {
                    titulo: title,
                    nombre: name,
                    apellido: lastName,
                    edad: age,
                    dpi: document,
                    birth: birth,
                    smoker: smoker,
                    med: medicamento,
                    pregnant: pregnant,
                    date:new Date().toLocaleDateString()
                })

            }


            return true;
       

        }
        catch(err){
            return false;
            
        }
       
    }


    
}
function CreateForm ({navigation, route}) {
    if(route){
        const [titulo,setTitulo] = useState('');
        const [nombre,setNombre] = useState('');
        const [apellido,setApellido] = useState('');
        const [edad,setEdad] = useState('');
        const [dpi,setDPI] = useState('');
        const [fecha,setFecha] = useState('');
        const [fumador,setFumador] = useState('');
        const [medicamento,setMedicamento] = useState('');
        const [embarazada,setEmbarazada] = useState('');
        const { name, email, photoUrl } = route.params;

        // const [textoBoton,setTextoBoton] = useState('Editar');
        const [buttonEnabled,setButtonEnabled] = useState(true);
        const isIOS = (Platform.OS === 'ios')? true : false;
        return(
            <View style={styles.container}>
           
            <KeyboardAvoidingView
            behavior={Platform.OS === "ios" ? "padding" : "padding"}
            style={styles.container}
            >
                <View style={styles.container}>
                    
                    <View style={styles.infoContainer}>
                        <ScrollView style={styles.infoScroll}>
                            <View style={styles.infoContainer2}>
                            <View style={styles.row}>
                                <Text>Titulo: </Text>
                            <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                    placeholder="Titulo formulario"  
                                    value={titulo} 
                                    autoCapitalize='none'
                                    editable={buttonEnabled}/>
                            </View>

                            <View style={styles.row}>
                            <Text>Nombre: </Text>
                            <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                placeholder="Nombre"  
                                value={nombre} 
                                autoCapitalize='none'
                                editable={buttonEnabled}/>

                            </View>
                            <View style={styles.row}>
                            <Text>Apellido: </Text>
                                <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                    placeholder="Apellido"  
                                    value={apellido} 
                                    autoCapitalize='none'
                                    editable={buttonEnabled}/>
                            </View>
                            <View style={styles.row}>
                            <Text>Edad: </Text>
                            <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                placeholder="Edad"  
                                value={edad}  
                                autoCapitalize='none'
                                editable={buttonEnabled}/>
                            </View>
    
                            <View style={styles.row}>
                            <Text>DPI: </Text>
                            <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                placeholder="DPI"  
                                value={dpi} 
                                autoCapitalize='none'
                                editable={buttonEnabled}/>
                            </View>
                            <View style={styles.row}>
                            <Text>F. nacimiento: </Text>
                            <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                placeholder="Fecha de nacimiento"  
                                value={fecha} 
                                autoCapitalize='none'
                                editable={buttonEnabled}/>
                            </View>
                            <View style={styles.row}>
                            <Text>Fumador: </Text>
                                <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                    placeholder="Fumador"  
                                    value={fumador} 
                                    autoCapitalize='none'
                                    editable={buttonEnabled}/>
                            </View>
                            <View style={styles.row}>
                            <Text>Medicamento: </Text>
                                <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                    placeholder="Consumes medicamento"  
                                    value={medicamento} 
                                    autoCapitalize='none'
                                    editable={buttonEnabled}/>
                            </View>
                            <View style={styles.row}>
                            <Text>NÃºmero: </Text>
                                <TextInput style={[styles.input,{color: (buttonEnabled)? 'black':'#c9c9c9'}]}
                                    placeholder="Embarazada"  
                                    value={embarazada}  
                                    autoCapitalize='none'
                                    editable={buttonEnabled}/>
                            </View>
                                
                            </View>
                            
                        </ScrollView>
                    </View>

                </View>

                <View style={styles.buttonArea}>
                    {/* <TouchableOpacity style={styles.shareButton} activeOpacity={0.7} onPress={() => {setButtonEnabled(!buttonEnabled); setTextoBoton((buttonEnabled)? "Editar":"Aceptar");}}>
                        <Text style={styles.shareButtonText}>{textoBoton}</Text>
                    </TouchableOpacity> */}

                    <TouchableOpacity style={styles.shareButton} >
                        <Text style={styles.shareButtonText}>Crear</Text>
                    </TouchableOpacity>

                </View>
                

                




            </KeyboardAvoidingView>
            
                
                
            </View>
        );
    }else{
        return(
            <View style={styles.container}>
                <Text style={styles.text}>No se ha encontrado el formulario</Text>
            </View>
        );
    }

}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor:"#f5f5f5",
        alignContent:"center",
        justifyContent:"flex-start",
        margin: 10,

    },
    shareButton:{
        marginTop:20,
        backgroundColor:"#143590",
        justifyContent:"center",
        alignItems: "center",
        width:"30%",
        height:50,
        alignSelf: "center",
        borderRadius:15,
        margin:20

    },
    shareButtonText:{
        color:"white",
    },
    header:{
        flex:0.1,
        backgroundColor:"#29abe2",
        width:'100%',
        flexDirection:"row",
        justifyContent:"flex-end",
        alignItems:"center"
      },
    content: {
        flex: 1,
        backgroundColor: 'lightblue'
    },
    footer: {
        flex:.04 ,
        backgroundColor: 'lightgrey'
    },
    formatexto:{
        flex: 1,
        fontSize:25,
        fontStyle: 'normal',
        fontWeight: 'bold',
        color:'#FFFF',
        textAlign: 'center'
 
    },
    formatexto2:{
        flex: 0.1,
        fontSize: 12,
        color:'black',
        textAlign: 'left',
        margin: 15
    },
    formatexto3:{
        flex: 1,
        fontSize:10,
        fontWeight: 'bold',
        color:'grey',
        textAlign: 'center',
        margin: 3
    },
    infoContainer:{
        flex:1,
        width:'100%',

    },infoContainer2:{
        flex:1,
        justifyContent: 'center',
        // backgroundColor:'red',
        width:'100%',

    },
    infoScroll:{
        flex: 1,
        backgroundColor:'#fefefe',
    }, input:{
        backgroundColor: '#fff',
        marginTop: 0,
        height: 40,
        borderBottomColor:"#143590",
        borderBottomWidth: 1,
        width:"50%",
        marginLeft: 5,
        marginTop: 10,
        
    },buttonArea:{
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
    },
    row:{
        flexDirection: 'row',
        alignItems: 'baseline',
        justifyContent: 'space-evenly',
        marginTop: 12,
        marginBottom: 12,
        paddingHorizontal: 10,
    },

});

export {saveFormInfo,CreateForm};
