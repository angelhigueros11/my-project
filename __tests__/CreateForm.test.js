import React from 'react';
import renderer from 'react-test-renderer';
import './Mocks/addEventListener.mock';
import {CreateForm, saveFormInfo} from '../components/CreateForm';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        dispatch: jest.fn(),
      }),
    };
});


describe('Pruebas en <CreateProfile />', () => {
    it('Renders correctly', () => {
        const name = 'Nombre';
        const email = 'Email';
        const photoUrl = 'Photo';
        const route = { params: {name,email,photoUrl}};
        const tree = renderer.create(<CreateForm route={route} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly withoutParams', () => {

        const tree = renderer.create(<CreateForm />).toJSON();
        expect(tree).toMatchSnapshot();        
    });

    it('saveFormInfo validData fireBase Error', () => {
        const navigation = { navigate: jest.fn() };
        let  email,title,name,lastName,document,smoker,medicamento,pregnant = 'data';
        let age = 20;
        let birth = '2020-01-01';
        expect(saveFormInfo(navigation,email,title,name,lastName,age,document,birth,smoker,medicamento,pregnant,true)).toBe(false);
        
    });

    it('saveFormInfo validData', () => {
        const navigation = { navigate: jest.fn() };
        let  email,title,name,lastName,document,smoker,medicamento,pregnant = 'data';
        let age = 20;
        let birth = '2020-01-01';
        expect(saveFormInfo(navigation,email,title,name,lastName,age,document,birth,smoker,medicamento,pregnant,false)).toBe(true);
    });

    it('saveFormInfo invalidData', () => {
        const navigation = { navigate: jest.fn() };
        let  email,title,name,lastName,age,birth,document,smoker,medicamento,pregnant = '';
        
        expect(saveFormInfo(navigation,email,title,name,lastName,age,document,birth,smoker,medicamento,pregnant,true)).toBe(false);
        
    });
});