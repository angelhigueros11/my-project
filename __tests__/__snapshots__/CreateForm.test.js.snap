// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CreateProfile /> Renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "margin": 10,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignContent": "center",
          "backgroundColor": "#f5f5f5",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "margin": 10,
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#f5f5f5",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "width": "100%",
          }
        }
      >
        <RCTScrollView
          style={
            Object {
              "backgroundColor": "#fefefe",
              "flex": 1,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                  "width": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Titulo: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Titulo formulario"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Nombre: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Nombre"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Apellido: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Apellido"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Edad: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Edad"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  DPI: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="DPI"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  F. nacimiento: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Fecha de nacimiento"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Fumador: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Fumador"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Medicamento: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Consumes medicamento"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Object {
                    "alignItems": "baseline",
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                    "marginBottom": 12,
                    "marginTop": 12,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text>
                  Número: 
                </Text>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  placeholder="Embarazada"
                  rejectResponderTermination={true}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "borderBottomColor": "#143590",
                        "borderBottomWidth": 1,
                        "height": 40,
                        "marginLeft": 5,
                        "marginTop": 10,
                        "width": "50%",
                      },
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#143590",
            "borderRadius": 15,
            "height": 50,
            "justifyContent": "center",
            "margin": 20,
            "marginTop": 20,
            "opacity": 1,
            "width": "30%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          Crear
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Pruebas en <CreateProfile /> Renders correctly withoutParams 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "margin": 10,
    }
  }
>
  <Text>
    No se ha encontrado el formulario
  </Text>
</View>
`;
